src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.WebAssembly.cs

```diff
@@ -119,6 +125,34 @@ private unsafe int JsIndexOfCore(ReadOnlySpan<char> source, ReadOnlySpan<char> t
             return idx;
         }
 
+        private char[] emptyCharsToRemove = {
+            '\u200d', '\u200b', '\u200c', '\uFEFF', '\u200E', '\u200F',
+            '\u2060', '\u2063', '\u2061', '\u2062', '\u2064', '\u180E',
+            '\u202A', '\u202B', '\u202D', '\u202E', '\u2066', '\u2067',
+            '\u2068', '\u2069', '\u202C'
+        };
+
+        private ReadOnlySpan<char> RemoveEmptyChars(ReadOnlySpan<char> source, CompareOptions options)
+        {
+            char[] result = new char[source.Length];
+            int resultIndex = 0;
+            foreach (char c in source)
+            {
+                if (Array.IndexOf(emptyCharsToRemove, c) == -1)
+                {
+                    result[resultIndex++] = c;
+                }
+            }
+            if ((options & CompareOptions.IgnoreCase) != 0)
+            {
+                string resultStr = new string(result, 0, resultIndex);
+                TextInfo textInfo = new CultureInfo(m_name).TextInfo;
+                resultStr = textInfo.ToLower(resultStr);
```

Comment:
`ToLowerInvariant` will ignore such cases as Turkish "i". We are doing a JS-based `ToLower` to keep these cases converted properly. Otherwise, we would not be able to keep non-invariant locales hashing work similar to ICU hashing.

Reply:
please make such comment to the code